Модуль наполнения базы данными из json файлов, и создание админа.

gorlovjob opened the pull request:
Модуль наполнения базы данными из json файлов, и создание суперпользователя.
mainapp\management\commands\fill_db.py - сам модуль.
mainapp\json\ - json файлы:
buyers.json - список пользователей.
categories.json - список категорий товаров.
items.json - список товаров.

Локально протестировано.
Запуск из командной строки: "python manage.py fill_db"

Косяк: если база уже создана, данные попадают в базу с ID не с нуля. Чтобы такого не происходило, нужно ввести настройки в саму базу (не помню какие, поэтому еще не делал).

gorlovjob committed:
Модуль наполнения базы данными из json файлов, и создание суперпользователя.

anoheroku commented on mainapp/management/commands/fill_db.py:
```diff
@@ -0,0 +1,83 @@
+
+# Загружаем из json файлов данные в базу
+def load_from_json(file_name):
+    with open(os.path.join(JSON_PATH, file_name + '.json'), 'r', encoding='utf-8') as infile:
```
Я бы к пути ещё добавил BASE_DIR из settings

anoheroku commented on mainapp/management/commands/fill_db.py:
```diff
@@ -0,0 +1,83 @@
+import os
+import json
+
+JSON_PATH = 'mainapp/json'
```
Жёстко задан прямой слэш

anoheroku commented on mainapp/management/commands/fill_db.py:
```diff
@@ -0,0 +1,83 @@
+        Buyer.objects.all().delete()
+
+        # Создаем суперпользователя при помощи менеджера модели
+        _superuser = Buyer.objects.filter(username='admin').first()
```
Мы только что всех удалили в строке 22

anoheroku commented on mainapp/management/commands/fill_db.py:
```diff
@@ -0,0 +1,83 @@
+        # Создаем суперпользователя при помощи менеджера модели
+        _superuser = Buyer.objects.filter(username='admin').first()
+        if not _superuser:
+            Buyer.objects.create_superuser('admin', 'buylist.project@gmail.com', 'y1u2i3o4y1u2i3o4')
```
Такие данные надо передавать через config.json, чтобы не светить пароль администратора

anoheroku reviewed (changes requested):
Жёстко задан прямой слэш в строке 6
Убрать данные суперпользователя из строки 27, изменить на config.SUPERUSER['username'] и т.п. (конфиг нужно подгрузить из settings)

anoheroku commented on mainapp/management/commands/fill_db.py:
```diff
@@ -0,0 +1,83 @@
+        for buyer in buyers:
+            buyers_objs.append(Buyer(**buyer))
+
+        Buyer.objects.bulk_create(buyers_objs)
```
в этом случае пароль не шифруется и залогиниться невозможно

gorlovjob committed:
Merge branch 'master' into gorlov

gorlovjob committed:
Исправлено:
- BASE_DIR из settings прописал в путь к json.
- Убрал прямой слеш из пути.
- Убрал проверку существования суперпользователя.
- Сделал еще один json файл - su.json. Там теперь можно вводить админов системы. Соответственно данные из кода убрал.
- Переделал добавление пользователей. Теперь их пароли шифруются и можно зайти по их данным в /api/v1/.

Commit:
Fill_db superuser data from config.json

Commit:
Only one superuser created
